lib/fiddle/ffi_backend.rb

```diff
@@ -433,7 +433,11 @@ def to_value
     end
 
     def inspect
-      "#<#{self.class.name} ptr=#{to_i.to_s(16)} size=#{@size} free=#{@free.inspect}>"
+      inspect_ptr = lambda do |ptr|
+        addr = ptr.to_i
+        addr == 0 ? "0x#{"0"*(SIZEOF_VOIDP * 2)}" : sprintf("%#0#{SIZEOF_VOIDP * 2 + 2}x", addr)
+      end
+      "#<#{self.class.name} ptr=#{inspect_ptr.(self)} size=#{@size} free=#{inspect_ptr.(@free)}>"
```

Comment:
How about using `0x%0.*x` instead of `inspect_ptr`?

```suggestion
      # SIZEOF_VOIDP * 2 == Math.log(2 ** (SIZEOF_VOIDP * 8), 16)
      pointer_inspect_width = SIZEOF_VOIDP * 2
      "#<%s ptr=0x%0.*x size=%d free=0x%0.*x>" % [
        self.class.name,
        pointer_inspect_width,
        to_i,
        size,
        pointer_inspect_width,
        @free || 0,
      ]
```

Reply:
Looks even better. Feel free to commit that instead.